it/v2/connectors/AuditingConnectorSpec.scala

```diff
@@ -135,4 +144,116 @@ class AuditingConnectorSpec
+                  .withHeader(Constants.XEoriHeader, equalTo(eori.map(_.value).getOrElse(None).toString))
+                  .withHeader(Constants.XURIPathHeader, equalTo("/customs/transits/movements"))
+                  .withHeader(Constants.XMessageTypeHeader, equalTo(messageType.map(_.code).getOrElse(None).toString))
+                  .withHeader(Constants.XMovementTypeHeader, equalTo(movementType.map(_.movementType).getOrElse(None).toString))
+                  .withHeader(Constants.XAuditSourceHeader, equalTo("common-transit-convention-traders"))
+                  .willReturn(aResponse().withStatus(ACCEPTED))
+              )
+
+              implicit val hc = HeaderCarrier(otherHeaders = Seq("path" -> "/customs/transits/movements"))
+              // when we call the audit service
+              val future = sut.post(
+                AuditType.DeclarationData,
+                contentType,
+                contentSize,
+                Source.empty,
+                movementId,
+                messageId,
+                eori,
+                movementType,
+                messageType
+              )
+
+              // then the future should be ready
+              whenReady(future) {
+                _ =>
+              }
+          }
+
+          "return a failed future if the audit message was not accepted" - Seq(BAD_REQUEST, INTERNAL_SERVER_ERROR).foreach {
+            statusCode =>
+              s"when a $statusCode is returned" in forAll(
+                Gen.option(arbitrary[EORINumber]),
+                Gen.option(arbitrary[MovementId]),
+                Gen.option(arbitrary[MessageId]),
+                Gen.option(arbitrary[MessageType]),
+                Gen.option(arbitrary[MovementType])
+              ) {
+                (eori, movementId, messageId, messageType, movementType) =>
+                  server.stubFor(
+                    post(
+                      urlEqualTo(targetUrl(AuditType.DeclarationData))
+                    )
+                      .withHeader(HeaderNames.AUTHORIZATION, equalTo(token))
+                      .withHeader(HeaderNames.CONTENT_TYPE, equalTo(contentType))
+                      .withHeader(Constants.XContentLengthHeader, equalTo(contentSize.toString))
+                      .withHeader(Constants.XMovementIdHeader, equalTo(movementId.map(_.value).getOrElse(None).toString))
+                      .withHeader(Constants.XMessageIdHeader, equalTo(messageId.map(_.value).getOrElse(None).toString))
+                      .withHeader(Constants.XEoriHeader, equalTo(eori.map(_.value).getOrElse(None).toString))
+                      .withHeader(Constants.XURIPathHeader, equalTo("/customs/transits/movements"))
+                      .withHeader(Constants.XMessageTypeHeader, equalTo(messageType.map(_.code).getOrElse(None).toString))
+                      .withHeader(Constants.XMovementTypeHeader, equalTo(movementType.map(_.movementType).getOrElse(None).toString))
```

Comment:
This is something that, as a team, we've been quite bad at in testing -- and wouldn't be caught here.

The problem we have here is that you've just replicated the line above and what it didn't catch is that the header value would be `None`, which is not useful. I suggest making a couple of tests, one that has the minimal set of headers, the maximum set of headers, a selection of headers, and then a test like the above. 

I get it's tough to test this with property based testing and to test it in a different way, but we do need to make sure it works as intended without just duplicating the actual code.